packages/server/routes/api/events.js

```diff
@@ -114,4 +160,91 @@ router.get(
   }
 );
 
+router.delete(
+  "/:eventId",
+  AuthController.authenticate,
+  eventIdValidator,
+  async (request, response) => {
+    const result = validationResult(request);
+
+    // if validation result is not empty, errors occurred
+    if (!result.isEmpty()) {
+      response.status(400).json({
+        status: "fail",
+        data: result.array(),
+      });
+      return;
+    }
+
+    try {
+      const { eventId } = request.params;
+      const deletedEvent = await EventController.deleteEvent(eventId);
+      if (deletedEvent) {
+        response.status(200).json({
+          status: "success",
+          data: null,
+        });
+      }
+    } catch (error) {
+      response.status(500).json({
+        status: "error",
+        message: error.message,
+      });
+    }
+  }
+);
+
+router.put(
+  //Route handler to update event
+  "/:eventId",
+  AuthController.authenticate,
+  updateEventValidator,
+  async (request, response) => {
+    // access validation results
+    const result = validationResult(request);
+
+    // if validation result is not empty, errors occurred
+    if (!result.isEmpty()) {
+      response.status(400).json({
+        status: "fail",
+        data: result.array(),
+      });
+      return;
+    }
+
+    try {
+      const { eventId } = request.params;
+      //See what data was given in request. Any data not given to be updated will be undefined
+      let givenData = {
+        title: request.body.title || undefined,
+        description: request.body.description || undefined,
+        startDate: request.body.startDate || undefined,
+        endDate: request.body.endDate || undefined,
+        location: request.body.location || undefined,
+        thumbnail: request.body.thumbnail || undefined,
+      };
```

Comment:
Update this to use the validated data from your validator. See [here](https://express-validator.github.io/docs/guides/getting-started#accessing-validated-data).